pkg/operator/resource/resourceapply/core.go

```diff
@@ -22,8 +22,10 @@ import (
 func ApplyNamespace(client coreclientv1.NamespacesGetter, recorder events.Recorder, required *corev1.Namespace) (*corev1.Namespace, bool, error) {
 	existing, err := client.Namespaces().Get(required.Name, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		actual, err := client.Namespaces().Create(required)
-		reportCreateEvent(recorder, required, err)
+		requiredCopy := required.DeepCopy()
+		actual, err := client.Namespaces().
+			Create(resourcemerge.WithCleanLabelsAndAnnotations(requiredCopy).(*corev1.Namespace))
+		reportCreateEvent(recorder, requiredCopy, err)
```

Comment:
report requiredCopy or actual or cleaned version?

Reply:
`actual` could be `nil` here in case of an error, so I guess the cleaned version should be reported. In C I would be sure that `requiredCopy == cleaned`, not sure if Go does additional magic to the pointer being passed to `WithCleanLabelsAndAnnotations` :thinking:

Reply:
Yes, they are equal

Reply:
I don't care much which one you print, just decide conciously :)